activesupport/lib/active_support/core_ext/time/calculations.rb

```diff
@@ -246,8 +246,10 @@ def minus_with_coercion(other)
   # Layers additional behavior on Time#<=> so that DateTime and ActiveSupport::TimeWithZone instances
   # can be chronologically compared with a Time
   def compare_with_coercion(other)
-    # we're avoiding Time#to_datetime cause it's expensive
-    if other.is_a?(Time)
+    # we're avoiding Time#to_datetime and Time#to_time because they're expensive
+    if other.class == Time
+      compare_without_coercion(other)
+    elsif other.is_a?(Time)
```

Comment:
:thought_balloon: this obviously looks ridiculous, but this second clause is meant to catch `ActiveSupport::TimeWithZone`, I believe. Please correct me if I'm wrong or if there's a better way to do this.